internal/injector/builtin/yaml/go-main.yml

```diff
@@ -21,7 +21,38 @@ aspects:
       - prepend-statements:
           imports:
             tracer: gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer
+            profiler: gopkg.in/DataDog/dd-trace-go.v1/profiler
+            log: log
+            os: os
+            slices: slices
+            strings: strings
           template: |-
             tracer.Start(tracer.WithOrchestrion(map[string]string{"version": {{printf "%q" Version}}}))
             defer tracer.Stop()
 
+            switch os.Getenv("DD_PROFILING_ENABLED") {
+            case "1", "true", "auto":
+              // The "auto" value can be set if profiling is enabled via the
+              // Datadog Admission Controller. We always turn on the profiler in
+              // the "auto" case since we only send profiles after at least a
+              // minute, and we assume anything running that long is worth
+              // profiling.
+              err := profiler.Start(
+                profiler.WithProfileTypes(
+                  profiler.CPUProfile,
+                  profiler.HeapProfile,
+                  // Non-default profiles which are highly likely to be useful:
+                  profiler.GoroutineProfile,
+                  profiler.MutexProfile,
+                ),
+                profiler.WithTags("orchestrion:true"),
+              )
+              if err != nil {
+                // TODO: is there a better reporting mechanism?
+                // The tracer and profiler already use the stdlib logger, so
+                // we're not adding anything new. But users might be using a
+                // different logger.
```

Comment:
That's a good question... I think it makes sense to do whatever `dd-trace-go` does... It may be interesting for it to expose a logger utility API that customers could re-configure to use their own logger instead? But I don't think this exists today... so... good enough!